lib/uring/uring.ml

```diff
@@ -413,17 +414,24 @@ let cancel t job user_data =
   ignore (Heap.ptr job : Uring.id);  (* Check it's still valid *)
   with_id t (fun id -> Uring.submit_cancel t.uring id (Heap.ptr job)) user_data
 
+(* Free stale entries in the sketch buffer, if possible.
+   This isn't quite right: a busy system might never have 0 unsubmitted entries.
+   We should probably track how many requests need to be submitted before each
+   of [t.sketch.old_buffers] can be released, but this will do for now. *)
+let gc_sketch t =
+  if Uring.sq_ready t.uring = 0 then Sketch.release t.sketch
+
 let submit t =
-  let v =
-    if t.dirty then begin
-      t.dirty <- false;
-      Uring.submit t.uring
-    end else
-      0
-  in
-  if Uring.sq_ready t.uring = 0 then
-    Sketch.release t.sketch;
-  v
+  if t.dirty then begin
```

Comment:
The rationale for checking `sq_ready` even if `dirty` was not set, is that technically you can still allocate things and never have `dirty` set, also more as of a safe guard in the future, checking `sq_ready` is cheap enough.

Reply:
OK, I've put it back how it was before.